samples/rnn.py

```diff
@@ -101,27 +88,42 @@ def _test(self):
         self.encoder.eval()
 
         with torch.no_grad():
+            macro_f1s = []
+            test_losses = []
             for input_tensor, target_tensor in logger.iterate("Test", self.test_loader):
                 encoder_hidden = self.encoder.init_hidden(self.device).double().to(self.device)
 
                 input_tensor = input_tensor.to(self.device).unsqueeze(1)
-                target_tensor = target_tensor.to(self.device).unsqueeze(1)
+                target_tensor = target_tensor.to(self.device).double()
 
                 encoder_output, encoder_hidden = self.encoder(input_tensor, encoder_hidden)
 
                 test_loss = self.loss(encoder_output, target_tensor)
+
                 macro_f1 = f1_score(y_true=target_tensor.cpu().detach().numpy().ravel(),
                                     y_pred=encoder_output.cpu().detach().to(torch.int32).numpy().ravel(),
                                     average='macro')
 
-            logger.store(test_loss=test_loss)
-            logger.store(accuracy=macro_f1)
+                test_losses.append(test_loss)
+                macro_f1s.append(macro_f1)
+
+            logger.store(test_loss=np.mean(test_losses))
```

Comment:
you can store `test_losses` directory, it will take the mean or produce a histogram